Sources/Public/Internal/CrashLoopDetector/BacktraceCrashLoopDetector.swift

```diff
@@ -0,0 +1,180 @@
+//
+//  BacktraceCrashLoopDetector.swift
+//  Backtrace
+//
+
+import Foundation
+
+@objc internal class BacktraceCrashLoopDetector: BacktraceCrashLoop {
+
+    internal struct StartUpEvent: Codable {
+        var uuid: String
+        var timestamp: Double
+        var crashesCount: Int
+        
+        func description() -> String {
+            let string = """
+                            New Crash Loop Event:\n
+                            UUID: \(uuid)\n
+                            Timestamp: \(timestamp)\n
+                            Crashes Count: \(crashesCount)
+                         """
+            return string
+        }
+    }
+
+    @objc private static let plistKey = "CrashLoopDetectorData"
+    @objc internal static let consecutiveCrashesThreshold = 5
+    @objc private(set) var consecutiveCrashesCount = 0
+
+    @objc private var threshold = consecutiveCrashesThreshold
+    
+    internal var startupEvents: [StartUpEvent] = []
+
+
+    @objc internal func updateThreshold(_ threshold: Int) {
+        self.threshold = threshold == 0 ? BacktraceCrashLoopDetector.consecutiveCrashesThreshold : threshold
+    }
+
+    @objc internal func detectCrashloop() -> Bool {
+
+        BacktraceCrashLoop.LogDebug("Starting Crash Loop Detection")
+        
+        loadEvents()
+        addStartupEvent()
+        saveEvents()
+
+        consecutiveCrashesCount = consecutiveEventsCount()
+
+        /*
+            true -> crash loop detected -> set safe mode
+            false -> crash loop NOT detected -> set normal mode
+         */
+        let result = consecutiveCrashesCount >= BacktraceCrashLoopDetector.consecutiveCrashesThreshold
+        BacktraceCrashLoop.LogDebug("Finishing Crash Loop Detection: \(result)")
```

Comment:
It is not important from the user perspective, maybe instead we can print something like "Is in the crash loop? \(result)" ?

Reply:
modified